src/lib/OS/SyscallHandler.cc

```diff
@@ -1,15 +1,711 @@
+      size_t length = info.registerArguments[1].get<size_t>();
+
+      int64_t result = munmap(addr, length);
+      // If successful, munmap returns the total number of bytes
+      // unmapped. If the value is greater than 0, 0 is returned as specified
+      // by the actual munmap specification. However, all negative values
+      // returned by munmap are in accordance with the munmap specification,
+      // so in case a negative value is returned it will remain the same.
+      result = result >= 0 ? 0 : result;
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {result}};
+      break;
+    }
+    case 222: {  // mmap
+      uint64_t addr = info.registerArguments[0].get<uint64_t>();
+      size_t length = info.registerArguments[1].get<size_t>();
+      int prot = info.registerArguments[2].get<int>();
+      int flags = info.registerArguments[3].get<int>();
+      int fd = info.registerArguments[4].get<int>();
+      off_t offset = info.registerArguments[5].get<off_t>();
+
+      uint64_t result = mmap(addr, length, prot, flags, fd, offset);
+      if (result <= 0) {
+        stateChange = {
+            ChangeType::REPLACEMENT, {info.ret}, {static_cast<int64_t>(-1)}};
+      } else {
+        stateChange = {ChangeType::REPLACEMENT, {info.ret}, {result}};
+      }
+      break;
+    }
+    case 226: {  // mprotect
+      // mprotect is not supported
+      // always return zero to indicate success
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {0ull}};
+      break;
+    }
+    case 235: {  // mbind
+      // mbind is not supported due to all binaries being single threaded.
+      // Always return zero to indicate success
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {0ull}};
+      break;
+    }
+    case 261: {  // prlimit64
+      // TODO: Functionality temporarily omitted as it is unused within
+      // workloads regions of interest and not required for their simulation
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {0ull}};
+      break;
+    }
+    case 278: {  // getrandom
+      // TODO: support flags argument
+
+      // seed random numbers
+      srand(clock());
+
+      // Write <buflen> random bytes to buf
+      uint64_t bufPtr = info.registerArguments[0].get<uint64_t>();
+      size_t buflen = info.registerArguments[1].get<size_t>();
+
+      char buf[buflen];
+      for (size_t i = 0; i < buflen; i++) {
+        buf[i] = (uint8_t)rand();
+      }
+
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {(uint64_t)buflen}};
+
+      stateChange.memoryAddresses.push_back({bufPtr, (uint8_t)buflen});
+      stateChange.memoryAddressValues.push_back(RegisterValue(buf, buflen));
+
+      break;
+    }
+    case 293: {  // rseq
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {0ull}};
+      break;
+    }
+
+    default:
+      break;
+  };
+
+  concludeSyscall(stateChange);
+}
+
+void SyscallHandler::readStringThen(char* buffer, uint64_t address,
```

Comment:
Moved translation to within function

Reply:
Given that we load the entire chunk, I think we should still do the translation outside the function but then skip the function call and return fatal if either of those faults occur. Both because if it's ignored and returns all 0 then no use of readStringThen can use a completely zero'ed out filename. Made this change, please review.